vivo-project/VIVO
# VCards in entry forms, list views, search indexing and visualization

**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues)**

Vitro PR

# What does this pull request do?
Added option to store participants as VCards for authorships, editorships and project participants.
Added option in editorship entry form to select organization as editor.
Added search indexing extension to index VCard participants.
Support visualization of VCard co-authors 

# What's new?
Modified entry forms:

- [AddAuthorsToInformationResourceGenerator.java](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-24828e4250b55f0bd838f2b4cf335ad0be4149856d104ce324b5535dca5a6f59)
[addAuthorsToInformationResource.ftl](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-7ac7aaa1813e85b9bce8800ee7af29892e0fcfd452e46a81f770e1e2bdde2de4)
[addAuthorsToInformationResource.js](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-98fa61faa54dc99ec927b0e1cb1f7b01ece763925bac2045fab7a74fbdaaf0e3)


- [AddEditorsToInformationResourceGenerator.java](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-4b674c4ebbcfb97820680db4977bd783f2eed4fe8ae71e865ca69e1ab9dd281c)
[addEditorsToInformationResource.ftl](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-6903fa17d6e69f5f38be0ed24f4ae9c152c03156f87ab6fd626b95976568a6b6)
[addEditorsToInformationResource.js](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-6932b147906da40aea93b38137090cffe159ac3dcee9fba836216e75efbaa988)

- [ProjectHasParticipantGenerator.java](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-47b050c9e3ce5791e9ff5503261effafef54d181654d19b5b9fc9400c0bfb8cf)
[projectHasParticipant.ftl](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-c4fe565f87beefb0a260c6028f8d5c3c0bb7d69ebbfa890be158898d63de70e4)

Created list view for presentation participant [listViewConfig-relatedExternalSpeakerRole.xml](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-bd42ade7cac72582396fc0b6478df91991782f326b9c4315e000ad65e052aa98)
Applied list view for participant faux property [PropertyConfig.n3](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-900c38f157ef62660ae2309859e95a712a0db6d5e1608cc707a33811bc4ef5fc)

Modified editorship list view to support VCards [listViewConfig-informationResourceInEditorship.xml](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-234844d4bc50259e97b512fa0ee38e05edf1430f864ca287e8620bb85fcce8cf) and [propStatement-informationResourceInEditorship.ftl](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-3611e6e63caa5ff247a91cfe1679658af6a8a2877a08baa76f31a8a2ee0bea2a)

VCard visualization modifications:
[Collaborator.java](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-9a5d7b4c2314be2b76e41c14703527f4bf7ee66db95fc2ae333d4aee5bd772a6)
[CollaborationDataViewHelper.java](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-081adcc6832f1148d23a05b780cadf5961453dedf6e649bbecb30ba6db182097)
[coAuthorPersonLevelD3.ftl](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-268a22d63ec57602f7b6098759008f66d7bf567f7272057dccb92c7edaaa764a)
[collaboratorToActivityCountTable.ftl](https://github.com/vivo-project/VIVO/compare/main...litvinovg:VIVO:vcards-original#diff-b69089529356d3691727d7e996b285658af45d96b14b463aaf20c3bbece13100)

# How should this be tested?
A description of what steps someone could take to:
* Try adding/removing participants to a publication by using authors, editors faux property:
* Add VCard participant
* Add person
* Select person from existing
* Add organization
* Select organization from existing
* For each case verify that added participant is listed in related list view
* Try adding participants to a project and presentations: vcard and real
* In person co-author network verify that vcard co-authors are visible if "Include external co-authors" is checked 
* Search for VCard participants

# Interested parties
@VIVO-project/vivo-committers

# Reviewers' expertise

Candidates for reviewing this PR should have some of the following expertises:
1. Java
2. HTML, CSS, JavaScript
3. FreeMarker
4. SPARQL
1. Natural language knowledge
    1. English
    2. German
    3. Spanish
    4. French
    5. Portuguese
    6. Russian
    7. Serbian

# Reviewers' report template
## General comment
A reviewer should provide here comments and suggestions for requested changes if any.
## Testing
A reviewer should briefly describe here how it was tested
## Code reviewing
A reviewer should briefly describe here which part was code reviewed
